src/apps/reservation-list/list/list.tsx

```diff
@@ -6,48 +6,82 @@ import { ReservationType } from "../../../core/utils/types/reservation-type";
 import ReservationMaterial from "../reservation-material/reservation-material";
 
 interface ListProps {
-  list: ReservationType[];
+  reservations: ReservationType[];
   header: string;
   emptyListLabel: string;
   pageSize: number;
 }
-const List: FC<ListProps> = ({ list, header, emptyListLabel, pageSize }) => {
+const List: FC<ListProps> = ({
+  reservations,
+  header,
+  pageSize,
+  emptyListLabel
+}) => {
+  const [dataReady, setDataReady] = useState<boolean>(false);
   const [displayedReservations, setDisplayedReservations] = useState<
     ReservationType[]
   >([]);
-  const { itemsShown, PagerComponent } = usePager(list.length, pageSize);
+  const { itemsShown, PagerComponent } = usePager(
+    reservations.length,
+    pageSize
+  );
 
   useEffect(() => {
-    if (list) {
+    if (reservations) {
       setDisplayedReservations(
-        getListItems(list, itemsShown) as ReservationType[]
+        getListItems(reservations, itemsShown) as ReservationType[]
       );
+      setDataReady(true);
     }
-  }, [itemsShown, list]);
+  }, [itemsShown, reservations]);
+
+  if (!dataReady) return <div />;
 
   return (
-    <>
-      <div className="dpl-list-buttons m-32">
-        <h2 className="dpl-list-buttons__header">
-          {header}
-          <div className="dpl-list-buttons__power">{list.length}</div>
-        </h2>
-      </div>
-      <ul className="list-reservation-container m-32">
-        {displayedReservations.length === 0 && (
-          <EmptyList emptyListText={emptyListLabel} />
-        )}
-        {displayedReservations.map((reservation) => (
-          <ReservationMaterial
-            key={reservation.identifier || reservation.faust}
-            identifier={reservation.identifier}
-            faust={reservation.faust}
-            reservation={reservation}
-          />
-        ))}
-        {PagerComponent}
-      </ul>
-    </>
+    <div>
```

Comment:
I think you should spilt this into two sub components. I find it too hard to read when using a huge ternary.
Example:
```tsx
{displayedReservations.length && <DisplayedReservations />}
{!displayedReservations.length && <EmptyDisplayedReservations />}

Reply:
done